gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -621,7 +623,11 @@ private void addHdfsJars(String hdfsJarFileList, Configuration conf) throws IOEx
   private void prepareJobInput(List<WorkUnit> workUnits) throws IOException {
     Closer closer = Closer.create();
     try {
-      ParallelRunner parallelRunner = closer.register(new ParallelRunner(this.parallelRunnerThreads, this.fs));
+      // Specify timeout on waiting for all workunits to be persisted as usually it could take long when
```

Comment:
Why not simply do parallelJobRunner.waitForTasks(Long.MAX_VALUE) in prepareJobInput? A pre-defined fixed timeout setting may not be ideal, since the timeout should ideally depend on the number of workunits created which may vary from run to run.